src/json/any.cr

```diff
@@ -120,14 +120,15 @@ struct JSON::Any
   # Traverses the depth of a structure and returns the value.
   # Returns `nil` if not found.
   def dig?(key : String | Int, *subkeys)
-    if (value = self[key]?) && value.responds_to?(:dig?)
-      value.dig?(*subkeys)
-    end
+    self[key]?.try(&.dig?(*subkeys))
```

Comment:
I'd prefer to keep `if value = self[key]? ...`  even without the second condition. Thats more readable than the proposed change.